src/chatbot.ts

```diff
@@ -66,6 +66,17 @@ const chatbot_helper = function (channel: string): Chatbot {
         // Called every time the bot connects to Twitch chat
         const onConnectedHandler = (addr: string, port: number) => {
           console.log(i18next.t("chatbotConnected", { addr, port }));
+          if (
+            !twitchApi.tokenScopes.includes("chat:edit") ||
+            !twitchApi.tokenScopes.includes("chat:read") ||
+            !twitchApi.tokenScopes.includes("moderator:read:chatters")
+          ) {
+            const err = i18next.t("requiredScopeError");
+            throw new Error(err);
+          }
+          if (!twitchApi.tokenScopes.includes("channel:read:subscriptions")) {
+            console.warn(i18next.t("subscribersScopeMissing"));
+          }
```

Comment:
I think this could just be done in `setup()` of `twitch-api.ts`, or does it need to be after the bot connected to chat specifically?
Maybe the `chat:edit` and `chat:read` checks could be done when calling `createTmiClient()`?